File: orig/Game/src/Explosion.java
import javax.microedition.lcdui.*;

public class Explosion extends GameObject
{
    int x, y, r;

    public Explosion(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public void update() {
	r += 2;
	if (r >= 20)
	    game.remove_object(this);
    }

    public void paint(Graphics g) {
        g.setColor(0xFFFFFF);
        g.drawArc(x-r, y-r, r*2, r*2, 0, 360);
    }
}

File: js/explosion.js
// Explosion.java port

import { GameObject } from './gameObject.js';

export class Explosion extends GameObject {
    constructor(game, x, y) {
        super(game);
        this.x = x;
        this.y = y;
        this.r = 0;
    }

    update() {
        this.r += 2;
        if (this.r >= 20)
            this.game.removeObject(this);
    }

    paint(g) {
        g.setColor(0xFFFFFF);
        g.drawArc(this.x - this.r, this.y - this.r, this.r * 2, this.r * 2, 0, 360);
    }
}
